model/service/BatteryService.php

```diff
@@ -72,6 +72,14 @@ public function deleteDeliveryFromBatteries(\core_kernel_classes_Resource $deliv
      */
     public function findDeliveryBattery(\core_kernel_classes_Resource $delivery);
 
+    /**
+     * Find all batteries without any delivery
+     *
+     * @return array|BatteryModel[]
+     * @throws BatteryException
+     */
+    public function findEmptyBatteries();
```

Comment:
This method will return all empty batteries. I think it will be better to have a method `deleteEmptyBattery`

Reply:
It makes sense, but before deleting we need to check assignments and assignments logic implemented in another extension. So we can not just remove all empty batteries without injecting code from taoNccerDelivery. But it will brake dependencies.

Reply:
make sense